From 83972b3d5fe76f43360cec758a90890a82b4ba16 Mon Sep 17 00:00:00 2001
From: Jose Dapena Paz <jdapena@igalia.com>
Date: Tue, 16 Jun 2020 17:01:54 +0200
Subject: Do not use NearbyConnection as parameter passed by copy

NearbyConnection is abstract, so it cannot be used as a function
parameter passed by copy.

Bug: 819294
Change-Id: I5021d09b075c79d50316c14418ee09f2a8465064
---
 .../browser/nearby_sharing/nearby_connections_manager.h  | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/chrome/browser/nearby_sharing/nearby_connections_manager.h b/chrome/browser/nearby_sharing/nearby_connections_manager.h
index 2c217d24820c..9d9fcabf32f4 100644
--- a/chrome/browser/nearby_sharing/nearby_connections_manager.h
+++ b/chrome/browser/nearby_sharing/nearby_connections_manager.h
@@ -26,9 +26,10 @@ class NearbyConnectionsManager {
   class IncomingConnectionListener {
     virtual ~IncomingConnectionListener() = default;
 
-    virtual void OnIncomingConnection(const std::string& endpoint_id,
-                                      const std::vector<uint8_t>& endpoint_info,
-                                      NearbyConnection connection) = 0;
+    virtual void OnIncomingConnection(
+        const std::string& endpoint_id,
+        const std::vector<uint8_t>& endpoint_info,
+        const std::unique_ptr<NearbyConnection>& connection) = 0;
   };
 
   // A callback for handling discovered devices while discovering.
@@ -79,7 +80,7 @@ class NearbyConnectionsManager {
   virtual void StopDiscovery() = 0;
 
   // Conntects to remote |endpoint_id| through Nearby Connections.
-  virtual NearbyConnection Connect(
+  virtual std::unique_ptr<NearbyConnection> Connect(
       std::vector<uint8_t> endpoint_info,
       const std::string& endpoint_id,
       base::Optional<std::vector<uint8_t>> bluetooth_mac_address,
-- 
2.20.1

